backend/aethel_db/views.py

```diff
@@ -79,55 +79,47 @@ def json_response(self) -> JsonResponse:
 
 class AethelQueryView(APIView):
     def get(self, request: HttpRequest) -> JsonResponse:
-        query_input = self.request.query_params.get("query", None)
-        if query_input is None or len(query_input) < 3:
-            return AethelListResponse().json_response()
+        word_input = self.request.query_params.get("word", None)
+        type_input = self.request.query_params.get("type", None)
 
-        def item_contains_query_string(item: LexicalItem, query_input: str) -> bool:
-            """
-            Checks if a LexicalItem contains a given input string in its word or its lemma.
-            """
-            return (
-                query_input.lower() in item.lemma.lower()
-                or query_input.lower() in item.word.lower()
-            )
+        # We only search for strings of 3 or more characters.
+        if word_input is not None and len(word_input) < 3:
+            return AethelListResponse().json_response()
 
         response_object = AethelListResponse()
 
-        # First we select all relevant samples from the dataset that contain the query string.
-        query_result = search(
```

Comment:
It turns out that this pre-filtering loop is not necessary, and we can improve performance slightly by removing it. I compared word and type searches with and without this step using a bigger dataset (3000 samples). Word searches performed slightly better without the pre-filtering (0.04s) step than with it (0.05s), and the same is true for type searches (0.52s with pre-filtering => 0.42s without).

Reply:
As an added benefit, we can drop a lot of code taken from `aethel` in `search.py`.